internal/testutil/test_hcls.go

```diff
@@ -0,0 +1,142 @@
+package testutil
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/google/uuid"
+)
+
+// ########################################
+// ###### RESOURCES CONSTs
+// ########################################
+
+const pfPort = "packetfabric_port"
+const pfCloudRouter = "packetfabric_cloud_router"
+const pfCloudRouterConnAws = "packetfabric_cloud_router_connection_aws"
+const pfCloudRouterBgpSession = "packetfabric_cloud_router_bgp_session"
+const pfCsAwsHostedConn = "packetfabric_cs_aws_hosted_connection"
+
+// ########################################
+// ###### HCLs FOR REQUIRED FIELDS
+// ########################################
+
+// packetfabric_port
+func HclPort() (hcl, resourceName, resourceReferece string) {
+	resourceReferece, resourceName = _generateResourceName(pfPort)
+	pop, _, media, availableSpeed, err := GetPopAndZoneWithAvailablePort(os.Getenv(PF_CRC_SPEED_KEY))
+	if err != nil {
+		log.Panic(err)
+	}
```

Comment:
How could you assert that pop in input is the pop in the API output and in the state? This part should be in the test code and passed as input to this function.

Reply:
@medzin, I tough about this a little bit more and I came with a solution that will enable us to validate those fields. I just pushed a new commit that will address this comment as well.